src/libs/Animals/TDynamicObject.h

```diff
@@ -13,12 +16,12 @@ class TDynamicObject
   public:
     TDynamicObject();
     virtual ~TDynamicObject();
-    void Initialize(const CVECTOR &_center, float _radius);
-    void Calculate(TDynamicObject **a, int aCount, TDynamicObject **d, int dCount, float _k = 1.0f);
+    void Initialize(const CVECTOR &_center, const float _radius);
+    void Calculate(std::vector<std::shared_ptr<TDynamicObject>> a, std::vector<std::shared_ptr<TDynamicObject>> d, const float _k = 1.0f);
```

Comment:
vector is passed by value, should be passed by reference I think, otherwise it is copied every time the function is called